coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -241,3 +235,65 @@ func (bp *BatchProverTask) recoverActiveAttempts(ctx *gin.Context, batchTask *or
 		log.Error("failed to recover batch active attempts", "hash", batchTask.Hash, "error", err)
 	}
 }
+
+func (bp *BatchProverTask) getBatchTaskDetail(ctx context.Context, dbBatch *orm.Batch, dbChunks []*orm.Chunk, chunkInfos []*message.ChunkInfo, chunkProofs []*message.ChunkProof) (*message.BatchTaskDetail, error) {
+	taskDetail := &message.BatchTaskDetail{
+		ChunkInfos:  chunkInfos,
+		ChunkProofs: chunkProofs,
+	}
+
+	if encoding.CodecVersion(dbBatch.CodecVersion) != encoding.CodecV3 && encoding.CodecVersion(dbBatch.CodecVersion) != encoding.CodecV4 {
+		return taskDetail, nil
+	}
+
+	chunks := make([]*encoding.Chunk, len(dbChunks))
+	for i, c := range dbChunks {
+		blocks, getErr := bp.blockOrm.GetL2BlocksInRange(ctx, c.StartBlockNumber, c.EndBlockNumber)
+		if getErr != nil {
+			return nil, fmt.Errorf("failed to get blocks in range for batch %d, chunk %d, (start: %d, end: %d): %w", dbBatch.Index, c.Index, c.StartBlockNumber, c.EndBlockNumber, getErr)
+		}
+		chunks[i] = &encoding.Chunk{Blocks: blocks}
+	}
+
+	dbParentBatch, getErr := bp.batchOrm.GetBatchByIndex(ctx, dbBatch.Index-1)
```

Comment:
imagine the first batch is codec v4 batch, index=1, if you get the parentBatch is like this, will get a error

Reply:
the parent batch (index=0) would be the genesis batch, which has a codec version 0. so no problem here.

Reply:
added a check: [2bd9438](https://github.com/scroll-tech/scroll/pull/1485/commits/2bd943860dafbed119f6076338da90d306b33ea9)